app/src/main/java/com/example/cookbook/ui/create_recipe/OwnRecipesViewModel.kt

```diff
@@ -24,22 +33,36 @@ class OwnRecipesViewModel @Inject constructor(private val recipeRepository: OwnR
     val isSuccessful: LiveData<Boolean>
         get() = _isSuccessful
 
+    private val composite = CompositeDisposable()
+
     init {
         observeRecipes()
     }
 
     private fun observeRecipes() {
-        _isLoading.value = true
-        viewModelScope.launch {
-            ownRecipes = recipeRepository.getRecipeListSync()
-            _isLoading.value = false
-        }
+        recipeRepository.getRecipeListSync()
+            .subscribeOn(schedulerIo)
+            .observeOn(schedulerMainThread)
+            .doOnSubscribe { _isLoading.value = true }
+            .subscribe(
+                {
+                    _ownRecipes.value = it
+                    _isLoading.value = ownRecipes.value?.isEmpty() == true
+                }, {
+                    Log.d("ERROR_LOG", it.localizedMessage ?: "unknown error")
```

Comment:
Do we need to add here `_isLoading.value = false` ?

Reply:
no, _isLoading.value depends on ownRecipes.value
ownRecipes.value can be empty or not, and if it's empty _isLoading.value = true, in other case _isLoading.value = false